libsolidity/formal/BMC.cpp

```diff
@@ -188,7 +188,8 @@ bool BMC::visit(FunctionDefinition const& _function)
 	{
 		reset();
 		initFunction(_function);
-		m_context.addAssertion(state().txTypeConstraints() && state().txFunctionConstraints(_function));
+		if (_function.isPublic())
+			m_context.addAssertion(state().txTypeConstraints() && state().txFunctionConstraints(_function));
```

Comment:
Looks good, but what else is covered by `txTypeConstraints`/`txFunctionConstraints`?

Reply:
type constraints on msg.*, tx.*, block.*, msg.value==0 is function is not payable, sets msg.sig and the first 4 bytes of msg.data with the function's signature.

Reply:
So basically here these constraints are not added to internal/external functions?

Reply:
I think external also goes in here. It shouldn't be added to `internal` functions.

Reply:
`	bool isPublic() const { return visibility() >= Visibility::Public; }`